src/ZendeskApi.Client/Resources/UploadResource.cs

```diff
@@ -1,40 +1,44 @@
 ﻿using System.Threading.Tasks;
-using ZendeskApi.Client.Http;
-using ZendeskApi.Client.Resources.ZendeskApi.Client.Resources;
 using ZendeskApi.Contracts.Models;
 using ZendeskApi.Contracts.Requests;
 using ZendeskApi.Contracts.Responses;
 
 namespace ZendeskApi.Client.Resources
 {
-    public class UploadResource : ZendeskResource<Upload>, IUploadResource
+    public class UploadResource : IUploadResource
     {
         private const string ResourceUri = "/api/v2/uploads";
+        private readonly IZendeskApiClient _apiClient;
 
-        public UploadResource(IRestClient client)
+        public UploadResource(IZendeskApiClient apiClient)
         {
-            Client = client;
+            _apiClient = apiClient;
         }
 
-        public void Delete(string token)
+        public Task DeleteAsync(string token)
         {
-            Delete($"{ResourceUri}/{token}");
-        }
-
-        public IResponse<Upload> Get(long id)
-        {
-            return Get<UploadResponse>($"{ResourceUri}/{id}");
+            using (var client = _apiClient.CreateClient(ResourceUri + "/"))
+            {
+                return client.DeleteAsync(token);
+            }
         }
 
         public async Task<IResponse<Upload>> GetAsync(long id)
         {
-            return await GetAsync<UploadResponse>($"{ResourceUri}/{id}").ConfigureAwait(false); ;
+            using (var client = _apiClient.CreateClient(ResourceUri + "/"))
+            {
+                var response = await client.GetAsync(id.ToString()).ConfigureAwait(false);
+                return await response.Content.ReadAsAsync<UploadResponse>();
+            }
         }
 
-        public IResponse<Upload> Post(UploadRequest request)
+        public async Task<IResponse<Upload>> PostAsync(UploadRequest request)
         {
-            var requestUri = Client.BuildUri(ResourceUri, $"filename={request.Item.FileName}{request.Token ?? ""}");
-            return Client.PostFile<UploadResponse>(requestUri, request.Item);
+            using (var client = _apiClient.CreateClient("/"))
+            {
+                var response = await client.PostAsJsonAsync($"{ResourceUri}?filename={request.Item.FileName}{request.Token ?? string.Empty}", request).ConfigureAwait(false);
+                return await response.Content.ReadAsAsync<UploadResponse>();
```

Comment:
`response.Content.ReadAsAsync` won't actually upload the file here, it's just getting a response for posting a file name.